src/cursor.rs

```diff
@@ -1,25 +1,50 @@
 use ffi;
 use statement::{State, Statement};
+use std::collections::HashMap;
+
 use {Result, Value};
 
 /// An iterator over rows.
 pub struct Cursor<'l> {
     state: Option<State>,
+    columns: Option<HashMap<String, usize>>,
     values: Option<Vec<Value>>,
     statement: Statement<'l>,
 }
 
+/// A row.
+#[derive(Debug)]
+pub struct Row {
+    columns: HashMap<String, usize>,
+    values: Vec<Value>,
+}
+
+/// A column index.
+///
+/// Indexed either by position (`usize`) or by column name (`&str`).
```

Comment:
I’ll after mering polish the docstrings, if you don’t mind.

Reply:
Of course not :sweat_smile: I know my english is far from perfect!

If you want to can use the "Add suggestion" feature of github to rewrite it? If you exceeds the line length I will wrap them after no worry